ldes-server-port-ingest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImpl.java

```diff
@@ -14,60 +14,61 @@
 
     private final MemberIngestValidator validator;
-	private final MemberRepository memberRepository;
-	private final ApplicationEventPublisher eventPublisher;
+    private final MemberRepository memberRepository;
+    private final ApplicationEventPublisher eventPublisher;
     private final MemberExtractorCollection memberExtractorCollection;
 
-	private static final Logger log = LoggerFactory.getLogger(MemberIngesterImpl.class);
+    private static final Logger log = LoggerFactory.getLogger(MemberIngesterImpl.class);
 
-	public MemberIngesterImpl(MemberIngestValidator validator, MemberRepository memberRepository,
+    public MemberIngesterImpl(MemberIngestValidator validator, MemberRepository memberRepository,
                               ApplicationEventPublisher eventPublisher, MemberExtractorCollection memberExtractorCollection) {
-		this.validator = validator;
-		this.memberRepository = memberRepository;
-		this.eventPublisher = eventPublisher;
+        this.validator = validator;
+        this.memberRepository = memberRepository;
+        this.eventPublisher = eventPublisher;
         this.memberExtractorCollection = memberExtractorCollection;
     }
 
-    private boolean ingestVersionObject(Member member) {
-        validator.validate(member);
-        final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
-        Optional<Member> ingestedMember = insertIntoRepo(member);
-        ingestedMember.ifPresentOrElse(insertedMember -> handleSuccessfulMemberInsertion(insertedMember, memberId),
-                () -> log.warn(DUPLICATE_MEMBER_INGESTED_MEMBER_WITH_ID_ALREADY_EXISTS, memberId)
-        );
-        return ingestedMember.isPresent();
-    }
-
     @Override
     public boolean ingest(String collectionName, Model ingestedModel) {
         final MemberExtractor memberExtractor = memberExtractorCollection
                 .getMemberExtractor(collectionName)
                 .orElseThrow(() -> new MissingResourceException("eventstream", collectionName));
-        final Member member = memberExtractor.extractMembers(ingestedModel).getFirst();
-        return ingestVersionObject(member);
+        final List<Member> members = memberExtractor.extractMembers(ingestedModel);
+        members.forEach(validator::validate);
+
+        return insertIntoRepo(members);
     }
 
-    private void handleSuccessfulMemberInsertion(Member member, String memberId) {
-        final var memberIngestedEvent = new MemberIngestedEvent(member.getModel(), member.getId(),
-                member.getCollectionName(), member.getSequenceNr());
+    private boolean insertIntoRepo(List<Member> members) {
+        members.forEach(Member::removeTreeMember);
+        List<Member> insertedMembers = memberRepository.insertAll(members);
+        if (insertedMembers.size() != members.size()) {
+            log.warn(DUPLICATE_MEMBERS_DETECTED);
+            return false;
+        }
+        insertedMembers.forEach(this::handleSuccessfulMembersInsertion);
```

Comment:
Can't we publish one event containing all ingested members? This allows the listener to process in bulk as well, improving performance. Currently, for the fragmentation, it does not matter as this is just a trigger. For the listener in retention, this would improve performance a lot.